jwst/ramp_fitting/tests/test_ramp_fit_cases.py

```diff
@@ -0,0 +1,541 @@
+import pytest
+import numpy as np
+import numpy.testing as npt
+
+from jwst.ramp_fitting.ramp_fit import ramp_fit
+from jwst.datamodels import MIRIRampModel
+from jwst.datamodels import GainModel, ReadnoiseModel
+
+#
+# Each of these 12 tests is for a single ramp in a single integration. The ramps
+#  have a variety of GROUPDQ vectors, with 1 or more segments in each ramp.  The
+#  comparison of the PRIMARY output results are partly to verify the slopes and
+#  variances of the combination of the segments in a ramp within the single
+#  integration.  The comparison of the OPTIONAL output results are to verify the
+#  results for each of the individual segments in a ramp.  Within each test is a
+#  description of classification ('CASE') within the code of all of the segments
+#  for the pixel based on the ramp's GROUPDQ, and the resulting segments and
+#  their SCI values (these are mostly for my reference).
+#
+
+def test_pix_0():
+    """
+    CASE A: segment has >2 groups, at end of ramp.
+    SCI seg is [15., 25., 35., 54., 55., 65., 75., 94., 95., 105.](A)
+    """
+
+    ngroups, nints, nrows, ncols, deltatime, gain, readnoise = set_scalars()
+    RampMod, RnMod, GMod, pixdq, groupdq, err = create_mod_arrays( ngroups,
+                           nints, nrows, ncols, deltatime, gain, readnoise )
+
+    # Populate pixel-specific SCI and GROUPDQ arrays
+    RampMod.data[0,:,0,0] = np.array([ 15., 25., 35., 54., 55.,
+                                    65., 75., 94., 95., 105.], dtype=np.float32)
+    RampMod.groupdq[0,:,0,0] = np.array([0, 0, 0, 0, 0, 0, 0, 0, 0, 0])
```

Comment:
I see that there aren't any non-zero DQ flags set for the ramp, so I guess you're NOT simulating CR hits, which means you can ignore my comment above.

Reply:
Ok. This is just a test for the DQ values, so the actual dummy SCI values up the ramp don't really matter.